File: 23-Java-Kotlin-Scala-Swift/23-Java-Kotlin-Scala-Swift/OrderProcessor.swift
struct Order { let id: String; let amount: Double }

actor OrderProcessor {
    // Swift Actors: Memory safety without GC pauses
    func process(order: Order) async {
        let converted = order.amount * 1.1
        print("Swift processed: \(order.id) -> \(converted)")
    }
}

File: 23-Java-Kotlin-Scala-Swift/23-Java-Kotlin-Scala-Swift/OrderProcessor.java
public record Order(String id, double amount) {}

public class OrderProcessor {
    public void process(Order order) {
        // Java 21+ Virtual Threads: Lightweight concurrency
        Thread.ofVirtual().start(() -> {
            double converted = order.amount() * 1.1; 
            System.out.println("Java processed: " + order.id() + " -> " + converted);
        });
    }
}
